scipy/ndimage/_measurements.py

```diff
@@ -618,82 +623,81 @@ def _stats(input, labels=None, index=None, centered=False):
     """
     def single_group(vals):
         if centered:
-            vals_c = vals - vals.mean()
-            return vals.size, vals.sum(), (vals_c * vals_c.conjugate()).sum()
+            vals_c = vals - xp.mean(vals)
+            return vals.size, xp.sum(vals), xp.sum(vals_c * vals_c.conjugate())
         else:
-            return vals.size, vals.sum()
+            return vals.size, xp.sum(vals)
 
     if labels is None:
         return single_group(input)
 
     # ensure input and labels match sizes
-    input, labels = np.broadcast_arrays(input, labels)
+    input, labels = xp.broadcast_arrays(input, labels)
 
     if index is None:
         return single_group(input[labels > 0])
 
-    if np.isscalar(index):
+    if xp_isscalar(index, xp):
```

Comment:
Doing input validation on `index` and then here simply `index.ndim == 0` looks more clear.